src/Sulu/Component/Content/Document/Subscriber/OrderSubscriber.php

```diff
@@ -46,52 +35,45 @@ public static function getSubscribedEvents()
     {
         return array(
             Events::PERSIST => 'handlePersist',
-            Events::HYDRATE => 'handleHydrate',
+            Events::METADATA_LOAD => 'handleMetadataLoad',
         );
     }
 
+    public function handleMetadataLoad(MetadataLoadEvent $event)
+    {
+        $metadata = $event->getMetadata();
+
+        if (false === $metadata->getReflectionClass()->isSubclassOf(OrderBehavior::class)) {
+            return;
+        }
+
+        $metadata->addFieldMapping('suluOrder', array(
+            'encoding' => 'system',
+            'property' => self::FIELD,
+        ));
+    }
+
     /**
      * @param PersistEvent $event
      */
     public function handlePersist(PersistEvent $event)
     {
-        $node = $event->getNode();
         $document = $event->getDocument();
 
-        if (false == $this->supports($document)) {
+        if (false == $document instanceof OrderBehavior) {
             return;
         }
 
-        $propertyName = $this->encoder->systemName(self::FIELD);
-
-        if ($node->hasProperty($propertyName)) {
+        // TODO: This does not seem quite right..
+        if ($document->getSuluOrder()) {
             return;
         }
 
+        $node = $event->getNode();
         $parent = $node->getParent();
         $nodeCount = count($parent->getNodes());
         $order = ($nodeCount + 1) * 10;
 
-        $node->setProperty($propertyName, $order, PropertyType::LONG);
-        $this->handleHydrate($event);
-    }
-
-    /**
-     * @param HydrateEvent $event
-     */
-    public function handleHydrate(AbstractMappingEvent $event)
```

Comment:
If I understand the context here, the removal of the hydrate/persist
methods where possible is the point of this PR.

By adding the metadata mapping hydrate and persist can be handled in the
"core" (and consequently, if we had a metadata cache, this subscriber
would not even be called).

On Mon, Jul 13, 2015 at 01:29:51AM -0700, Daniel Rotter wrote:
>    In [1]src/Sulu/Component/Content/Document/Subscriber/OrderSubscriber.php:
> 
>  >          $parent = $node->getParent();
>  >          $nodeCount = count($parent->getNodes());
>  >          $order = ($nodeCount + 1) * 10;
>  >
>  > -        $node->setProperty($propertyName, $order, PropertyType::LONG);
>  > -        $this->handleHydrate($event);
>  > -    }
>  > -
>  > -    /**
>  > -     * @param HydrateEvent $event
>  > -     */
>  > -    public function handleHydrate(AbstractMappingEvent $event)
> 
>    Why can handleHydrate be removed.
> 
>    —
>    Reply to this email directly or [2]view it on GitHub.
> 
>    Reverse link: [3]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/sulu-io/sulu/pull/1332#discussion_r34441965
>    2. https://github.com/sulu-io/sulu/pull/1332/files#r34441965
>    3. https://github.com/sulu-io/sulu/pull/1332/files#r34441965